app/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_status_validation_mixin.py

```diff
@@ -56,3 +60,308 @@ def test_update_experiment_with_invalid_publish_status_error(self):
         )
         self.assertFalse(serializer.is_valid())
         self.assertIn("experiment", serializer.errors)
+
+    @parameterized.expand(
```

Comment:
These are all the same test contents, but I broke it out into a few different ones by the `Lifecycle` it was testing because it was easier to debug. I'd love thoughts on whether I should make it one (huge) test 🤔

Reply:
I am happy with the split tests as the test name focuses on functionality, but I don't mind having huge test too, up to you

Reply:
Hrmmmm.....  So yes I like this, and if we test the combinations of statuses here to stress test the allowed status/publish_status transitions, then we don't need more than one test cover that in the actual serializer test above, just a single one to test integration is sufficient and then we can rigorously test all the combinations here and then we're covered.

As far as whether to make these separate or make a single test with a giant list of all the combinations.....  I feel like I could go either way on it.  On the one hand, the test body is the same in all of them so there's no real reason to make them separate, but then the parameter list gets MASSIVE.  Is a massive list of status combinations okay?  Could be?  Makes it very clear which combinations are allowed and which aren't.  I dunno, I could go either way on it.  

If you keep them separate then I would take `status` out of the parameter list since then it's redundant.  And I would use `status` directly instead of bothering with the lifecycles since then it's really clear what exact combination is being talked about.

Reply:
When I was writing these, it was really helpful to keep them split up to debug them. With the parameterized tests, it just fails everything and you have to test each parameter line individually to see which was the failing line. 

I like the idea of using `status` directly -- that makes it much more clear